src/main/haskell/kore/src/Kore/Step/BaseStep.hs

```diff
@@ -802,13 +816,21 @@ stepWithRemainders
         ( OrStepResult level variable
         , StepProof level variable
         )
-stepWithRemainders tools substitutionSimplifier patt rules =
-    stepWithRemaindersForUnifier
-        tools
-        (UnificationProcedure unificationProcedure)
-        substitutionSimplifier
-        (map (\ (RewriteRule rule) -> rule) rules)
-        patt
+stepWithRemainders tools substitutionSimplifier patt rules
+  = do
+    resultOrError <- runExceptT
+        $ stepWithRemaindersForUnifier
+            tools
+            (UnificationProcedure unificationProcedure)
+            substitutionSimplifier
+            (map (\ (RewriteRule rule) -> rule) rules)
+            patt
+    case resultOrError of
+        Left _ -> error $
```

Comment:
Do I understand correctly, that this is a "not implemented" case? It would be clearer if the error message had those words in it.